lib/api/hub_config_api.rb

```diff
@@ -1,20 +1,23 @@
 class HubConfigApi
   require 'cgi'
-
-  HUB_HOST = Rails.configuration.hub_config_host
   HEALTHCHECK_ENDPOINT = 'service-status'.freeze
   CERTIFICATES_ROUTE = '/config/certificates/'.freeze
   CERTIFICATE_ENCRYPTION_ENDPOINT = "%{entity_id}/certs/encryption".freeze
   CERTIFICATES_SIGNING_ENDPOINT = "%{entity_id}/certs/signing".freeze
 
-  def initialize; end
+  def initialize(environment: :test)
+    @environment = environment
+    @header = hub_environment(:'secure-header') == 'true'
+    @hub_host = hub_environment(:'hub-config-host')
+    @service_authentication_header = ENV['SELF_SERVICE_AUTHENTICATION_HEADER']
```

Comment:
```suggestion
    @service_authentication_header = ENV.fetch('SELF_SERVICE_AUTHENTICATION_HEADER', nil)
```
I would be tempted not to make it break if it's not present. Since it might not be needed.